mlflow/models/evaluation/default_evaluator.py

```diff
@@ -258,6 +259,45 @@ def _do_plot(**kwargs):
 }
 
 
+def _evaluate_custom_metric(custom_metric, eval_df, builtin_metrics):
+    result = custom_metric(eval_df, builtin_metrics)
+    if result is None:
+        _logger.warning(
+            f"Custom metric function '{custom_metric.__name__}' returned None. Logging ignored."
+        )
+        return None, None
+
+    def __validate_metrics_dict(metrics):
+        return isinstance(metrics, dict) and all(
+            isinstance(metric_name, str) and isinstance(metric_val, (int, float, np.number))
+            for metric_name, metric_val in metrics.items()
+        )
+
+    def __validate_artifacts_dict(artifacts):
+        return isinstance(artifacts, dict) and all(
+            isinstance(artifacts_name, str) for artifacts_name in artifacts.keys()
+        )
+
+    if isinstance(result, dict) and __validate_metrics_dict(result):
+        return result, None
+
+    if (
+        isinstance(result, tuple)
+        and __validate_metrics_dict(result[0])
+        and __validate_artifacts_dict(result[1])
+    ):
+        return result
+
+    raise MlflowException(
+        f"Custom metric '{custom_metric.__name__}' did not return in an expected format."
+        f"The two acceptable return types are:"
+        f"1. Dict[AnyStr, Union[int, float, np.number]: a dictionary of metrics"
+        f"2. Tuple[Dict[AnyStr, Union[int, float, np.number]], Dict[AnyStr, Any]]: a"
+        f"   dictionary of metrics and a dictionary of artifacts."
+        f"For more, check out the docstring for mlflow.evaluate"
```

Comment:
```suggestion
        f"Custom metric '{custom_metric.__name__}' did not return an expected format."
```

Reply:
Is it bad practice to have long exception messages? Are users expected to refer to the docstring?

Reply:
Just seemed like a duplication of information that could get annoying to maintain since it's not in an obvious place.
I think we can instead link them to the actual docs once they're published.

Reply:
Make sense!